### ISSUE TITLE
Browser bundle contains NodeJS code that throws "Uncaught TypeError: Cannot read properties of undefined (reading 'split')"

### ISSUE THREAD

**Issue by bitflower** [Automattic/mongoose#11943]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

6.3.8

### Node.js version

16.14.0

### MongoDB server version

4.2.19

### Description

When bundling up `mongoose` for the browser this code is included from the file `dist/browser.umd.js`:

```javascript
e.nodeMajorVersion=parseInt(t.versions.node.split(".")[0],10);
```

It throws this error in Chrome: 

```
Uncaught TypeError: Cannot read properties of undefined (reading 'split')
```

My guess is that the code line above was never supposed to end up in the browser bundle because it is server-side node code.

### Steps to Reproduce

See the `browser.umd.js` file in the `dist` folder of `mongoose`.

### Expected Behavior

The `browser.umd.js` contains only browser-compatible code.

**Comment by vkarpov15** [comment]
What are you using to bundle Mongoose? I confirmed this code doesn't appear using Mongoose's top-level `webpack.config.js`: https://github.com/Automattic/mongoose/blob/b66cdc62feb0ac735158ea4aee71cd9236481758/webpack.config.js .

It looks like the code you're referencing comes from [this file in js-bson](https://github.com/mongodb/js-bson/blob/96130212f1e675015c717d98e1d8470de3120375/etc/prepare.js), which is a post-install script for the MongoDB driver's bson parser.

**Comment by bitflower** [comment]
I use StencilJS which uses Rollup in the background. So you say it depends on my bundler? I just thtought it is Mongoose specific because I found the problematic line in your bundle. But maybe rollup doesn‘t take the bundled browser bundle?

**Comment by vkarpov15** [comment]
That's strange, we'll check to see if that code is in the v6.3.8 bundle. Running Webpack on current master branch doesn't result in that code in the bundle...

**Comment by vkarpov15** [comment]
I'm unable to repro, below shows my terminal output from running `npm install mongoose@6.3.8` and searching for the code in question.

```
$ npm install mongoose@6.3.8

added 28 packages, and audited 29 packages in 4s

4 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
$ grep -r "nodeMajor" node_modules/mongoose/*
$ 

```

If this issue is still affecting you, please provide more detailed repro instructions.

**Comment by alexredder** [comment]
@bitflower did you solve it? .. i have the same problem and cant find a solution?

**Comment by Vitalik-Hakim** [comment]
Thank

> I am also experiencing this issue with mongoose 6.5.5 on node 16.17.
> 
> I can reproduce with a simplified example:
> 
> ```
> ❯ cat package.json
> {
>   "name": "mongoose-error",
>   "version": "1.0.0",
>   "dependencies": {
>     "mongoose": "~6.5.5"
>   }
> }
> ```
> 
> ```
> ❯ npm i
> 
> added 28 packages, and audited 29 packages in 820ms
> 
> 4 packages are looking for funding
>   run `npm fund` for details
> 
> found 0 vulnerabilities
> ```
> 
> ```
> ❯ grep -r "nodeMajor" node_modules/mongoose/*
> node_modules/mongoose/dist/browser.umd.js:e.noop=function(){},e.errorToPOJO=function(t){if(!(t instanceof Error))throw new Error("`error` must be `instanceof Error`.");var e,r={},o=n(Object.getOwnPropertyNames(t));try{for(o.s();!(e=o.n()).done;){var i=e.value;r[i]=t[i]}}catch(t){o.e(t)}finally{o.f()}return r},e.nodeMajorVersion=parseInt(t.versions.node.split(".")[0],10)}).call(this,r(10))},function(t,e,r){"use strict";(function(t){
> ```
> 
> Downgrading to mongoose 6.5.2 removes the nodeMajorVersion reference.

Thanks! it fixed it

**Comment by ames20** [comment]
I've had the same issue (Node 18.10.0, Mongoose 6.6.4 (and 6.6.3), Mongodb 4.10.0). The nodeMajorVersion is still in the dist/browser.umd.js file and the only fix is to downgrade to Mongoose@6.5.2.

Any chance of correcting the reference in the current version? 

Thanks so much

**Comment by bitflower** [comment]
Sorry for me not getting back sooner. I think I have a workaround replacing the problemativ parts with `rollup` during bundling.

@vkarpov15 might be open to re-open this one with the number of people in here bumping up. I could go ahead and create a Stencil + Mongoose repo repro.

**Comment by bitflower** [comment]
@alexredder See above comment

**Comment by dlogic-lbs** [comment]
Yo, I'm having the same issue with browser.umd.js - node v16.14.0 and mongoose 6.6.3
Would much appreciate a solution to this issue, please.

**Comment by vkarpov15** [comment]
Fix will be in v6.6.7 shortly, see #12577

**Comment by clathey123** [comment]
Has this issue been resolved????  If so, how do I implement the latest release of mongoose?  

Thank you!   Cheryl
